### ISSUE TITLE
Persistent Fail on `ORDER_NOT_OPEN` OperationalException

### ISSUE THREAD

**Issue by robmoggach** [gcarq/freqtrade#385]
## Environment

  * Python Version: Python 3.6.3
  * Branch: Develop
  * Last Commit ID: 1fe86656e168a256c237e55a013fbc22ac75b1c2
 
## Problem

Bot crashes and can't be restarted. 

### Steps to reproduce:

1. Notice stalled trade with profit that doesn't match bittrex
   `121  BTC_XCP   1 d      0.23%`
2. Try `/forcesell 121` multiple times without success
3. Delete order manually on bittrex hours later.
4. Bot crashes and cannot be restarted.

### Output

>Status: Got OperationalException:
```python
Traceback (most recent call last):
  File "/home/freqtrade/freqtrade/freqtrade/main.py", line 104, in _process
    check_handle_timedout(_CONF['unfilledtimeout'])
  File "/home/freqtrade/freqtrade/freqtrade/main.py", line 154, in check_handle_timedout
    exchange.cancel_order(trade.open_order_id)
  File "/home/freqtrade/freqtrade/freqtrade/exchange/__init__.py", line 150, in cancel_order
    return _API.cancel_order(order_id)
  File "/home/freqtrade/freqtrade/freqtrade/exchange/bittrex.py", line 197, in cancel_order
    order_id=order_id))
freqtrade.OperationalException: ORDER_NOT_OPEN params=(03b02exx-adxx-4xx9-xxfc-xx258XXX26xx)
```
> Issue /start if you think it is safe to restart.

**Comment by glonlas** [comment]
Thank you for this great bug report. 

I think you can fix your DB by closing the trade with the following query (backup your db file before `cp tradesv3.sqlite tradesv3.sqlite.bak`):
```sql
UPDATE trades SET open_rate=0, open_order_id=NULL WHERE id=<TRADE_ID>;
```

SQL helper: https://github.com/gcarq/freqtrade/blob/develop/docs/sql_cheatsheet.md

Let me know if it helps.

**Comment by robmoggach** [comment]
For anyone seeing this in future, that SQL should show the `TRADE_ID` in quotes and should be specific to be `TRADE_UUID`.

Also ... that didn't fix the problem. It still crashes with the same exception.

**Comment by stephendade** [comment]
I may have fixed this issue. Turns out that check_handle_timedout runs over completed orders sometimes - I added a check for that.

**Comment by jblestang** [comment]
Maybe we could add on startup a sync between bittrex and the database to avoid this kind of issues?

**Comment by jblestang** [comment]
COuld we catch the exception and uuid and clean the database accordingly?

**Comment by glonlas** [comment]
I will see what I can do, because this issue is very annoying.

**Comment by glonlas** [comment]
Will be fixed with PR#429
